index.js

```diff
@@ -38,12 +39,21 @@ Cloneable.prototype.clone = function () {
   }
 
   this._clonesCount++
-  return pump(this, new Clone(this))
+  return this.pipe(new Clone(this))
+}
+
+function forwardDestroy (src, dest) {
+  src.on('error', destroy)
+  src.on('close', destroy)
+
+  function destroy (err) {
+    dest.destroy(err)
+  }
```

Comment:
First thing I tried. But it results in premature close errors, because when one stream (either the source or cloneable) emits `end`, the other (the cloneable or clone, respectively) might not be fully consumed yet - so it's unsafe to call `destroy` on it.